data_model/src/block.rs

```diff
@@ -325,10 +313,81 @@ mod candidate {
         payload: BlockPayload,
     }
 
+    #[derive(Decode, Deserialize)]
+    struct BlockPayloadCandidate {
+        header: BlockHeader,
+        transactions: Vec<CommittedTransaction>,
+        event_recommendations: Vec<EventBox>,
+    }
+
+    #[derive(Decode, Deserialize)]
+    struct BlockHeaderCandidate {
+        height: NonZeroU64,
+        prev_block_hash: Option<HashOf<SignedBlock>>,
+        transactions_hash: HashOf<MerkleTree<SignedTransaction>>,
+        creation_time_ms: u64,
+        view_change_index: u32,
+        consensus_estimation_ms: u64,
+    }
+
+    impl BlockHeaderCandidate {
+        const fn creation_time(&self) -> Duration {
+            Duration::from_millis(self.creation_time_ms)
+        }
+
+        fn validate(self) -> Result<BlockHeader, &'static str> {
+            #[cfg(feature = "std")]
+            if std::time::Instant::now().elapsed() < self.creation_time() {
```

Comment:
I'm a bit concerned that block validity now depends on when block was received.

Reply:
i don't think it compromise security, but might harm liveliness if peers clocks drift from each other.

Reply:
This is upper bound. What do you think about having the lower bound? I can only define lower bound if I define both lower and upper. Because if lower bound is defined and block time is set far into the future no new blocks can be produced